site/src/docs/components/login/api.mdx

```diff
@@ -0,0 +1,774 @@
+
+Custom modules may emit any other kind of signals with different payloads.
+
+##### Emitting signals
+
+Only custom modules can emit signals. See the <InternalLink href="/components/login/customisation#emitting-signals">custom module documentation</InternalLink>.
+
+See also <InternalLink href="/components/login/customisation#custom-namespaced-beacons-for-modules">custom namespaced beacons</InternalLink>.
+
+#### Important
+
+If a listener is added while signalling and it is triggered by the currently emitted signal, a loop could be created. This could occur when using hooks and not memoizing listeners. Avoid emitting signals inside listeners.
+
+##### Listening to signals
+
+A signal listener is a function which receives one argument: the signal. A listener can listen to all signals or just one type or signal with a certain namespace.
+Listeners can be even more specific and listen to signals with certain payloads. In short, a listener can listen to any properties of the signal and is triggered when all properties match.
+
+The listener is called only if the emitted signal matches the given props.
+
+For example, if the trigger props (first argument) passed to `beacon.addListener(trigger, listener)` is `{ type:'error' }`, the listener (second argument) is called when the emitted signal has a matching type. It does not matter what other props the signal has.
+
+If the trigger props are `{ namespace:'myModule', payload:{ type: 'click' } }`, the emitted signal must have those properties with the same, exact values. Other properties are not checked.
+
+The trigger can also be a function. Internally all triggers are converted to functions.
+
+Instead of writing signal types as strings, it is better to use predefined triggers:
+
+- <InternalLink href="/components/login/api#dedicated-signal-triggers">Oidc client triggers</InternalLink>.
+- <InternalLink href="/components/login/api#dedicated-signal-triggers-1">Api tokens client triggers</InternalLink>.
+- <InternalLink href="/components/login/api#dedicated-signal-triggers-2">Session poller triggers</InternalLink>.
+- <InternalLink href="/components/login/api#dedicated-signal-triggers">Generic triggers</InternalLink>.
+
+##### Generic triggers
+
+Generic `create...` functions create trigger functions from given props. Creator functions are not objects, so they can be used to create triggers for multiple namespaces, payloads and so on.
+
+| Trigger creator                                                  | Arguments                                                                                 | Default        |
+| ---------------------------------------------------------------- | ----------------------------------------------------------------------------------------- | -------------- |
+| `createTriggerPropsForAllSignals(namespace)`                     | Namespace to listen to. If omitted listens to all signals                                 | all namespaces |
```

Comment:
Dot missing at the end of sentence "If omitted listens to all signals"

Reply:
Fixed.

Made some helper functions to find missing dots 😄 

```
function getTableCellsWithoutPunctuation(cellCaption){

    const tables = document.querySelectorAll("table");
return Array
      .from(tables).map(table=>{
	const captionIndex = Array.from(table.querySelectorAll("thead th")).map(el=>el.innerText).indexOf(cellCaption)
	const rows = Array.from(table.querySelectorAll("tbody tr"))
	return rows.map(row=>Array.from(row.querySelectorAll("td")).filter((el,i)=>i === captionIndex))[0]
	}).map(el=>el[0] ? el[0].innerText : null).filter(text=>text && text.substr(-1) !== ".")
    
    
}

function getPWithoutPunctuation(){


return Array
      .from(document.querySelectorAll("#content p, #content li")).map(el=>el.innerText).filter(text=>text && text.substr(-1).match(/[\.-:;]/) === null)
    
    
}
```